keras_nlp/models/roberta/roberta_classifier.py

```diff
@@ -134,43 +182,98 @@ def from_preset(
+
+        Raw string inputs.
         ```python
-        input_data = {
-            "token_ids": tf.ones(shape=(1, 12), dtype=tf.int64),
-            "padding_mask": tf.constant(
-                [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0], shape=(1, 12)
-            ),
-        }
+        # Create a dataset with raw string features in an `(x, y)` format.
+        features = ["The quick brown fox jumped.", "I forgot my homework."]
+        labels = [0, 3]
+
+        # Create a RobertClassifier and fit your data.
+        classifier = keras_nlp.models.RobertClassifier.from_preset(
+            "roberta_base",
+            num_classes=4,
+        )
+        classifier.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        )
+        classifier.fit(x=features, y=labels, batch_size=2)
+        ```
+
+        Raw string inputs with customized preprocessing.
+        ```python
+        # Create a dataset with raw string features in an `(x, y)` format.
+        features = ["The quick brown fox jumped.", "I forgot my homework."]
+        labels = [0, 3]
+
+        # Use a shorter sequence length.
+        preprocessor = keras_nlp.models.RobertPreprocessor.from_preset(
+            "roberta_base",
+            sequence_length=128,
+        )
 
-        # Load backbone architecture and weights from preset
-        classifier = keras_nlp.models.RobertaClassifier.from_preset(
+        # Create a RobertClassifier and fit your data.
+        classifier = keras_nlp.models.RobertClassifier.from_preset(
             "roberta_base",
             num_classes=4,
+            preprocessor=preprocessor,
         )
-        output = classifier(input_data)
+        classifier.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+        )
+        classifier.fit(x=features, y=labels, batch_size=2)
+        ```
 
-        # Load randomly initialized model from preset architecture
-        classifier = keras_nlp.models.RobertaClassifier.from_preset(
+        Preprocessed inputs.
+        ```python
+        # Create a dataset with preprocessed features in an `(x, y)` format.
+        preprocessed_features = {
+            "token_ids": tf.ones(shape=(2, 12), dtype=tf.int64),
+            "padding_mask": tf.constant(
+                [[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]] * 2, shape=(2, 12)
+            ),
+        }
+        labels = [0, 3]
+
+        # Create a RoBERTa classifier and fit your data.
+        classifier = keras_nlp.models.RobertClassifier.from_preset(
             "roberta_base",
-            load_weights=False,
             num_classes=4,
+            preprocessor=None,
+        )
+        classifier.compile(
+            loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
         )
-        output = classifier(input_data)
+        classifier.fit(x=preprocessed_features, y=labels, batch_size=2)
         ```
         """
+        if preset not in cls.presets:
```

Comment:
I think just leave this error message at the end of the block for now. No point having it twice.